homeassistant/components/yolink/sensor.py

```diff
@@ -71,8 +74,10 @@ class YoLinkSensorEntityDescription(
         native_unit_of_measurement=PERCENTAGE,
         name="Battery",
         state_class=SensorStateClass.MEASUREMENT,
-        value=lambda value: percentage.ordered_list_item_to_percentage(
-            [1, 2, 3, 4], value
+        value=lambda value: (
+            percentage.ordered_list_item_to_percentage([1, 2, 3, 4], value)
+            if value > 0
```

Comment:
Please don't use ternary operator expression if it spans more than one line. I suggest extracting the lambda to a normal function.